tt_metal/common/core_coord.h

```diff
@@ -20,6 +20,14 @@ using std::pair;
 
 using CoreCoord = tt_xy_pair;
 
+// Adding to tt::tt_metal namespace as we transition to moving this out of global namespace eventually.
+namespace tt::tt_metal {
+   using ::CoreCoord;
+}
+
+namespace ttnn::tensor {
```

Comment:
tt_metal is a lower level library and should know nothing about ttnn

Reply:
@eyonland 's suggestion ( a temp one) until CoreCoord and other necessary classes are moved out of global namespace. are you suggesting we remove namespace ttnn::tensor and keep namespace tt::tt_metal for now ?

Reply:
@KalaivaniMCW  we for sure don't want to do this here.

My suggestion is to move this namespace declaration that defines ttnn::tensor::CoreCoord to ttnn/cpp/ttnn/tensor/types.hpp